pkg/webhook/webhook.go

```diff
@@ -36,6 +37,23 @@ type combinedDrainData struct {
 	InstanceID string
 }
 
+var instanceEventMap = make(map[string]*monitor.InterruptionEvent)
+var m sync.RWMutex
+
+type PostDecorator func(additionalInfo ec2metadata.NodeMetadata, event *monitor.InterruptionEvent, nthConfig config.Config)
+
+func SinglePost(postFn PostDecorator) PostDecorator {
+	return func(additionalInfo ec2metadata.NodeMetadata, event *monitor.InterruptionEvent, nthConfig config.Config) {
+		m.Lock()
+		defer m.Unlock()
+
+		if previousEvent, ok := instanceEventMap[event.InstanceID]; !ok || (previousEvent.Kind == monitor.RebalanceRecommendationKind && event.Kind == monitor.SpotITNKind) {
+			postFn(additionalInfo, event, nthConfig)
+			instanceEventMap[event.InstanceID] = event
```

Comment:
1. Since `postFn` returns no indication of success or failure the `instanceEventMap` can be updated before calling `postFn`, and then the mutex can be unlocked before calling `postFn` to avoid blocking other workers unnecessarily.
2. Are there other cases besides RR+SpotITN were NTH should send two webhooks for a single instance id?
3. Entries in the map need to eventually be removed to avoid over consuming memory
4. Since only the `Kind` field of previous events is used, just store that.